common/src/main/java/com/wynntils/features/DiscordRichPresenceFeature.java

```diff
@@ -59,33 +61,89 @@ public void onXpChange(SetXpEvent event) {
 
     @SubscribeEvent
     public void onWorldChange(WorldStateEvent event) {
+        if (!Services.Discord.isReady()) return;
+
         if (displayWorld.get()) {
             switch (event.getNewState()) {
                 case WORLD -> {
                     Services.Discord.setState(Models.WorldState.getCurrentWorldName());
-                    checkTerritory();
+                    startTerritoryCheck();
                 }
                 case HUB, CONNECTING -> {
                     Services.Discord.setDetails("");
                     Services.Discord.setState("In Hub");
                     Services.Discord.setWynncraftLogo();
+                    stopTerritoryCheck();
                 }
                 case CHARACTER_SELECTION -> {
                     if (displayLocation.get()) {
                         Services.Discord.setDetails("Selecting a character");
                     }
                     Services.Discord.setState("");
                     Services.Discord.setWynncraftLogo();
+                    stopTerritoryCheck();
                 }
             }
         } else {
             Services.Discord.setState("");
+            stopTerritoryCheck();
         }
     }
 
     @SubscribeEvent
     public void onDisconnect(ConnectionEvent.DisconnectedEvent e) {
         Services.Discord.unload();
+        stopTerritoryCheck();
+    }
+
+    @Override
+    protected void onConfigUpdate(Config<?> config) {
+        tryUpdateDisplayedInfo();
+    }
+
+    @Override
+    public void onEnable() {
+        // This isReady() check is required for Linux to not crash on config change.
+        if (!Services.Discord.isReady()) {
+            // Load the Discord SDK
+            if (!Services.Discord.load()) {
+                // happens when wrong version of GLIBC is installed and Discord SDK fails to load
+                Managers.Feature.crashFeature(this);
+            }
```

Comment:
Despite the name, this does not stop the code from running, it just adds the feature to the list of crashed features. So you'll need a return here so as not fall through to tryUpdateDisplayedInfo(). For now it seems like it should be okay since there are guards in that method, but that seems to be more by incident than design, so any change in that method might mess things up.